pkg/actuators/awsplacementgroup/controller.go

```diff
@@ -3,91 +3,622 @@ package awsplacementgroup
+		equal = reflect.DeepEqual(pg.Status.ObservedConfiguration, *pg.Spec.ManagementSpec.Managed)
+	}
+
+	if equal {
+		condition.Status = metav1.ConditionTrue
+		condition.Reason = AWSPlacementGroupConfigurationInSyncConditionReason
+	} else {
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = AWSPlacementGroupConfigurationMismatchConditionReason
+	}
+
+	meta.SetStatusCondition(&pg.Status.Conditions, condition)
+}
+
+// validateAWSPlacementGroup validates an AWSPlacementGroup configuration.
+func validateAWSPlacementGroup(pg *machinev1.AWSPlacementGroup) error {
+	// First validation should happen via webhook before the object is persisted.
+	// This is a complementary validation to fail early in case of lacking proper webhook validation.
+	switch pg.Spec.ManagementSpec.ManagementState {
+	case machinev1.ManagedManagementState, machinev1.UnmanagedManagementState:
+		// valid values
+	default:
+		return fmt.Errorf("invalid AWSPlacementGroup. spec.managementSpec.managementState must either be %s or %s",
+			machinev1.ManagedManagementState, machinev1.UnmanagedManagementState)
+	}
+
+	if pg.Spec.ManagementSpec.ManagementState == machinev1.ManagedManagementState {
+		if pg.Spec.ManagementSpec.Managed == nil {
+			return fmt.Errorf("invalid AWSPlacementGroup. spec.managementSpec.managed must not be nil when spec.managementSpec.managementState is %s",
+				machinev1.ManagedManagementState)
+		}
+
+		// A Managed placement group may be moved to Unmanaged, however an Unmanaged
+		// group may not be moved back to Managed.
+		if pg.Status.ManagementState == machinev1.UnmanagedManagementState {
+			return fmt.Errorf("invalid AWSPlacementGroup. spec.managementSpec.managementState cannot be set to %s once it has been set to %s",
+				machinev1.ManagedManagementState, machinev1.UnmanagedManagementState)
+		}
+	}
+
+	if pg.Spec.ManagementSpec.Managed != nil {
+		switch pg.Spec.ManagementSpec.Managed.GroupType {
+		case machinev1.AWSClusterPlacementGroupType, machinev1.AWSPartitionPlacementGroupType, machinev1.AWSSpreadPlacementGroupType:
+			// valid values
+		default:
+			return fmt.Errorf("invalid AWSPlacementGroup. spec.managementSpec.managed.groupType must either be %s, %s or %s",
+				machinev1.AWSClusterPlacementGroupType, machinev1.AWSPartitionPlacementGroupType, machinev1.AWSSpreadPlacementGroupType)
+		}
+
+		if pg.Spec.ManagementSpec.Managed.Partition != nil {
+			if pg.Spec.ManagementSpec.Managed.Partition.Count < 1 || pg.Spec.ManagementSpec.Managed.Partition.Count > 7 {
+				return fmt.Errorf("invalid AWSPlacementGroup. spec.managementSpec.managed.partition.count must be greater or equal than 1 and less or equal than 7")
+			}
+		}
+	}
+
+	return nil
+}
+
+// requeueIn returns the time.Duration that represents the amount
+// of time before to wait before requeuing.
+// If the computed time.Duration is negative,
+// meaning the provided time.Time is in the past, it returns 0s.
+func requeueIn(at time.Time) time.Duration {
```

Comment:
Should this be `requeueAt`, given we are passing it a time at which we want it to then requeue? I would expect `requeueIn` to take a duration, not a time